src/main/java/randoop/generation/DemandDrivenInputCreation.java

```diff
@@ -0,0 +1,489 @@
+   * objects. This method consider boxing equivalence when comparing boxed and unboxed types.
+   *
+   * @param t the type of objects to be included in the subset
+   * @return a list of sequences that contains only the objects of the specified type and their
+   *     sequences
+   */
+  private static SimpleList<Sequence> getSequencesForTypeConsideringBoxing(
+      SequenceCollection sequenceCollection, Type t) {
+    Set<Sequence> subPoolOfType = new HashSet<>();
+    Set<Sequence> sequences = sequenceCollection.getAllSequences();
+    for (Sequence seq : sequences) {
+      if (EquivalenceChecker.areEquivalentTypesConsideringBoxing(
+          seq.getLastVariable().getType().getRuntimeClass(), t.getRuntimeClass())) {
+        subPoolOfType.add(seq);
+      }
+    }
+    SimpleList<Sequence> subPool = new SimpleArrayList<>(subPoolOfType);
+    return subPool;
+  }
+
+  /**
+   * Get a set of classes that are utilized by the demand-driven input creation process but were not
+   * explicitly specified by the user. As of the current manual, Randoop only invokes methods or
+   * constructors that are specified by the user. Demand-driven input creation, however, ignores
+   * this restriction and uses all classes that are necessary to generate inputs for the specified
+   * classes. This methods returns a set of unspecified classes to help inform the user of the
+   * classes that are automatically included in the testing process.
+   *
+   * @return A set of unspecified classes that are automatically included in the demand-driven input
+   *     creation process.
+   */
+  public static Set<Class<?>> getUnspecifiedClasses() {
+    return unspecifiedClasses;
+  }
+
+  /**
+   * Determines whether the set of unspecified classes is empty.
+   *
+   * @return true if the set of unspecified classes is empty, false otherwise.
+   */
+  public static boolean isUnspecifiedClassEmpty() {
+    return unspecifiedClasses.isEmpty();
+  }
+
+  /**
+   * Get a set of classes that are utilized by the demand-driven input creation process but were not
+   * explicitly specified by the user. This method filters out classes that are part of the Java
+   * standard library.
+   *
+   * @return A set of unspecified, non-Java classes that are automatically included in the
+   *     demand-driven input creation process.
+   */
+  public static Set<Class<?>> getNonJavaClasses() {
+    Set<Class<?>> nonJavaClasses = new LinkedHashSet<>();
+    for (Class<?> cls : unspecifiedClasses) {
+      if (!startsWithJava(cls.getName()) && !cls.isPrimitive()) {
+        nonJavaClasses.add(cls);
+      }
+    }
+    return nonJavaClasses;
+  }
+
+  /**
+   * Determines whether a class name starts with "java.".
+   *
+   * @param className the name of the class
+   * @return true if the class name starts with "java.", false otherwise.
+   */
+  public static boolean startsWithJava(String className) {
+    return className.startsWith("java.") || className.matches("^\\[+.java\\..*");
```

Comment:
Precompute the regular expression into a `Pattern` and store it in a local variable, for efficiency.